### DISCUSSION TITLE
YoloV5 vs resnet50 is very big

### DISCUSSION THREAD

**Discussion by mgabell** [ultralytics/yolov5#12659] [General]
Hi,

I have been running resnet50 and want to increase speed for Jetson. I have raised this question before but training with the standard parameters for yolo and resnet (not changes to hyper params, only batch size , image input size and epochs).
After some testing with about 10 classes with transfer learning from a yolov5m its very clear that Yolo is extremely poor cases where there are one class dominating and it gives a lot of false positives.

For example, I have one class with 10'000 labels and another with only 5-600 labels. The second class is quite similar in shape but still very distinct to the eye. Resnet picks about 75% while yolo on the same dataset and training material catch about 19%.

My question is, which parameters should I start tweaking for yolo? IoU perhaps?? I have looked at the parameters available but not very sure about the effect each parameter will have...

Any help would do...

**Comment by NavodPeiris** [comment]
you should go for MobileNet models if you want model to be light weight and less resource intensive.

also checkout this repo: https://github.com/Navodplayer1/MobileNet_96x96_greyscale_weights

**Reply by mgabell** [reply]
We have run MobilNet but its was not good enough either, or rather not fast enough on Jetson NX. So we went for yolo on Orin with Deepstream and it was speed wise much faster, So our hope is that Yolo is as good as we have seen in some cases on one class. But due to over weight of one class compared to another I think we need need to work on both data set and hyper parameters but its hard to understand which to start meddling with.

Secondly the Mobilnet we ran was with tensorflow 1.15 and we need to keep up with tensorflow versions and it was a big problem to get the later version to work on NX. So best hope now is Yolo (super easy to train and deploy once the gstreamer is in place). If this does not work then we should look into deploying resnet50 and convert to mobilNet and run with deepstream. OpenCV is also too slow for some reason. Too much overhead or not utilizing the memory on the Jetson properly so Deepstream is the thing to use...
